### PR TITLE
Make configparser.RawConfigParser use SectionProxy where appropriate

### PR DESCRIPTION
**Pull request by OddBloke**
This reflects the code in the cpython tree and makes the following
(valid) code type-check correctly:

```
from configparser import ConfigParser

config = ConfigParser()
config.read_dict({'section': {'key': 'false'}})
assert config['section'].getboolean('key') is False
```

### CODE REVIEW

**Comment by OddBloke** [comment]
Also relevant to getting the types in check: https://github.com/python/cpython/pull/3012

File: stdlib/3/configparser.pyi [hunk 1]
```diff
@@ -99,8 +99,11 @@ class RawConfigParser(_parser):
     def get(self, section: str, option: str, *, raw: bool = ..., vars: _section = ..., fallback: str = ...) -> str:  # type: ignore
         ...
 
-    # This is incompatible with Mapping so we ignore the type.
-    def items(self, section: str = ..., raw: bool = ..., vars: _section = ...) -> Iterable[Tuple[str, _section]]: ...  # type: ignore
+    @overload
+    def items(self, *, raw: bool = ..., vars: _section = ...) -> AbstractSet[Tuple[str, SectionProxy]]: ...
+
+    @overload
+    def items(self, section: str, raw: bool = ..., vars: _section = ...) -> Iterable[Tuple[str, str]]: ...
```
Comment (JelleZijlstra) [inline stdlib/3/configparser.pyi]: It actually returns a list. Any reason not to use `List` in the annotation?

Comment (OddBloke) [inline stdlib/3/configparser.pyi]: No reason I can think of; updated.

**Comment by reddraggone9** [comment]
I created a file `test.py` with the code in the OP and added the following line to the end:
```
reveal_type(config['section'].getboolean('key'))
```
Using this branch of typeshed, mypy gives me
```
test.py:6: error: Revealed type is 'Any'
```
If I understand this correctly, using the [current definition for SectionProxy][0] causes `getboolean()` to fall back on `__getattr__()`, which is a very loose definition since [`SectionProxy` pulls][1] its get methods [from the parser][2] that created it so that [parser subclasses have a nicer time][3]. I tried out mypy for the first time today and ran into this while trying to get `--strict` to pass on the file I was messing around with, so sorry if I'm missing something.

[0]: https://github.com/python/typeshed/blob/7fe417ca582011dd2bf54f36502ef48529ffc6d8/stdlib/3/configparser.pyi#L134-L148
[1]: https://github.com/python/cpython/blob/957d0e9b59bd27ca7c473560634d8b5dbe66338c/Lib/configparser.py#L1223-L1226
[2]: https://github.com/python/cpython/blob/957d0e9b59bd27ca7c473560634d8b5dbe66338c/Lib/configparser.py#L1286-L1303
[3]: https://bugs.python.org/issue18159

Bottom line, it seems unfortunate that mypy doesn't know that `getboolean()` returns a `bool`. Can we fix that?

**Comment by OddBloke** [comment]
@reddraggone9 That is something that needs fixing, but AFAICT this PR doesn't regress that behaviour. 

Indeed, without this PR, I get the following output from mypy:

```
test.py:5: error: Mapping[str, str] has no attribute "getboolean"
test.py:6: error: Revealed type is 'Any'
test.py:6: error: Mapping[str, str] has no attribute "getboolean"
```

File: stdlib/3/configparser.pyi [hunk 2]
```diff
@@ -99,8 +99,11 @@ class RawConfigParser(_parser):
     def get(self, section: str, option: str, *, raw: bool = ..., vars: _section = ..., fallback: str = ...) -> str:  # type: ignore
         ...
 
-    # This is incompatible with Mapping so we ignore the type.
-    def items(self, section: str = ..., raw: bool = ..., vars: _section = ...) -> Iterable[Tuple[str, _section]]: ...  # type: ignore
+    @overload
+    def items(self, *, raw: bool = ..., vars: _section = ...) -> AbstractSet[Tuple[str, SectionProxy]]: ...
```
Comment (ambv) [inline stdlib/3/configparser.pyi]: In general, return types that are as concrete as possible, and accept types in arguments that are as generic as possible.

**Comment by OddBloke** [comment]
@ambv Done.
